src/stepfunctions/steps/service.py

```diff
@@ -887,3 +892,66 @@ def __init__(self, state_id, **kwargs):
                                                                    ElasticMapReduceApi.ModifyInstanceGroupByName)
 
         super(EmrModifyInstanceGroupByNameStep, self).__init__(state_id, **kwargs)
+
+
+class StepFunctionsStartExecutionStep(Task):
+
+    """
+    Creates a Task state that starts an execution of another state machine. See `Manage AWS Step Functions Executions as an Integrated Service <https://docs.aws.amazon.com/step-functions/latest/dg/connect-stepfunctions.html`_ for more details.
+
+    Property flags: There are three flags (wait_for_callback, wait_for_completion and async_call) that can be set in order to select which Step Functions resource to use.
+        One of three must be enabled to create the step successfully.
+    """
+
+    def __init__(self, state_id, wait_for_callback=False, wait_for_completion=True, async_call=False, **kwargs):
+        """
+        Args:
+            state_id (str): State name whose length **must be** less than or equal to 128 unicode characters. State names **must be** unique within the scope of the whole state machine.
+            wait_for_callback(bool, optional): Boolean value set to `True` if the Task state should wait for callback to resume the operation. (default: False)
+            wait_for_completion (bool, optional): Boolean value set to `True` if the Task state should wait to complete before proceeding to the next step in the workflow. (default: True)
+            async_call(bool, optional): Boolean value set to `True` for the Task proceed to the next step in the workflow without waiting for completion. (default: False):
```

Comment:
This would be a good opportunity to improve as proposed here: https://github.com/aws/aws-step-functions-data-science-sdk-python/pull/74

Using a single enum arg is easier to reason about instead of three mutually exclusive args, which either overwrite each other or raise an Error if more than one are provided.

Reply:
I'm in favour of an enum as well (which is what we also do in the CDK) - the tradeoff is it might be dissimilar to other service integrations.

i.e. if any of the existing service integrations add support for all 3 modes of operation, we would likely have some inconsistency between service integrations.

not sure that it's a problem we need to futureproof for, but just mentioning it for completeness

Reply:
Also in favour of an enum even though it introduces inconsistency with the other service integrations
This will be clearer and easier to use for our customers

Reply:
None of the service integrations today support all 3 patterns, which is why this is a good time to introduce this change.

`ecs:runTask` does support all 3, but this SDK only exposes `.sync` and request-response. 

- https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/src/stepfunctions/steps/compute.py#L184-L197
- https://docs.aws.amazon.com/step-functions/latest/dg/connect-ecs.html

Reply:
this is the enum we added in the [CDK](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.IntegrationPattern.html)
and Python specific [documentation](https://docs.aws.amazon.com/cdk/api/latest/python/aws_cdk.aws_stepfunctions/IntegrationPattern.html) for it

keep in mind that you'd want to name the options in the enum in a way that's idiomatic for existing data science SDK customers.

Reply:
We have an enum already https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/23878de4ba909564569b90d9ab3ae689cacf04dd/src/stepfunctions/steps/integration_resources.py#L20

Reply:
I was planning to use the the existing enum mentioned by @wong-a 
Change to use enum arg instead of 3 flags will be included in the next commit

Reply:
i should clarify:

wasn't suggesting adding a new enum, was only sharing the CDK enum to indicate that there shouldn't be a new entry in the enum to surface `sync:2` - it's just what `WaitForCompletion` maps to for this service integration.

> None of the service integrations today support all 3 patterns, which is why this is a good time to introduce this change.

That's true, but my comment around futureproofing is more around how you'd introduce it if a service integration that currently supports 2 patterns starts supporting all 3.

i.e. if EMR aded wait for completion, since we have already exposed `wait_for_completion` as a property:
* we could introduce the enum as input allowing for all 3 integration types
  * we would essentially map 2 properties as ways in which wait_for_completion could be requested (enum and flag)
  * we would need to add validation that enum and this flag don't conflict with each other

I still prefer the enum because it's clear and explicit about behaviour that will be observed.

Reply:
>wasn't suggesting adding a new enum, was only sharing the CDK enum to indicate that there shouldn't be a new entry in the enum to surface sync:2 - it's just what WaitForCompletion maps to for this service integration.

Yes, i was planning to remove the `WaitForCompletionWithJsonResponse` that was added in this PR to not expose the response format to the customer (this was discussed [here](https://github.com/aws/aws-step-functions-data-science-sdk-python/pull/166#discussion_r706701234))

Reply:
After reflexion, I opted to add a new enum so that we have the same terms than the ones described in [Service Integration Patterns](https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html) similarly to the CDK doc mentioned by @shivlaks (details in PR description)

Reply:
>i.e. if EMR aded wait for completion, since we have already exposed wait_for_task_token as a property:
> - we could introduce the enum as input allowing for all 3 integration types
>   - we would essentially map 2 properties as ways in which wait_for_task_token could be requested (enum and flag)
>   - we would need to add validation that enum and this flag don't conflict with each other
> I still prefer the enum because it's clear and explicit about behaviour that will be observed.
> If that situation materialized, I'd also recommend marking the flag as deprecated and dropping support in v3 to normalize every integration to using the enum.

This seems like an acceptable solution when adding support for a 3rd integration pattern
Mapping the existing flag to the enum and marking that flag as "to be deprecated" will ensure backward compatibility and give a heads up to our customers

For consistency, when adding support for a new service integrations, regardless of the number of integration patterns, I also think we should use the enum solution if we decide to go that route